lsp-go.el

```diff
@@ -47,8 +47,93 @@ completing function calls."
   :type '(repeat string)
   :group 'lsp-gopls)
 
+(defcustom lsp-gopls-build-flags '("-tags")
```

Comment:
Lists must be vectors otherwise it wont work when using emacs 27 native json serialization which threats all lists as objects. We should update that in the settings generator.

Reply:
:+1: 

Is there a better `:type` to use for vectors?

Reply:
Yes, we could use:


```
 :type '(restricted-sexp :match-alternatives (lambda (xs)	  :type '(repeat string)
                                                (and	
                                                 (vectorp xs)	
                                                 (seq-every-p #'stringp xs))))
```

This must be done in the whole project, not only in this PR.

Reply:
Ah, very good. I'll make a note to start tidying some of those up next.

Reply:
Please if you want to work on a particular task, file an issue and ask me to assign it to you to avoid collisions.